src/Options.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Minishlink\WebPush;
+
+class Options
+{
+    /**
+     * @var array
+     */
+    private $allowed = ['ttl', 'urgency', 'topic'];
```

Comment:
I am not a big fan of hardcoded values.

Reply:
> That’s a lot of changes! The code look cleaner and should be easier to maintain.
> I made some comments and I will be happy to discuss if you disagree with them or if I was not clear enough.

Thanks!! I really appreciate the feedback and help working through this PR as a whole! 😃

Reply:
I agree, I think I was trying to support BC initially with this... going to mull over named arguments instead.

Reply:
Ahh, I remember now - according to the spec, there are quite a few more options available. Enough so that passing them as named parameters through a constructor would blow out the method signature.

This may not be the best implementation given that criteria, so open to suggestions and/or changes. But let me know if that context changes things for you.